processor/src/main/java/org/mapstruct/ap/internal/util/Message.java

```diff
@@ -70,6 +70,7 @@
     PROPERTYMAPPING_INVALID_PARAMETER_NAME( "Method has no source parameter named \"%s\". Method source parameters are: \"%s\"." ),
     PROPERTYMAPPING_NO_PROPERTY_IN_PARAMETER( "The type of parameter \"%s\" has no property named \"%s\"." ),
     PROPERTYMAPPING_INVALID_PROPERTY_NAME( "No property named \"%s\" exists in source parameter(s). Did you mean \"%s\"?" ),
+    PROPERTYMAPPING_INVALID_PROPERTY_NAME_SOURCE_HAS_NO_PROPERTIES( "No property named \"%s\" exists in source parameter(s). Type \"%s\" has no properties." ),
```

Comment:
`has no properties`? That's not certain, is it?

Reply:
We are certain. `mostSimilarWord` can be `null` only if there were no other `similarWords`

https://github.com/mapstruct/mapstruct/blob/857f87276f686f0659b0fa604aa363e25701916b/processor/src/main/java/org/mapstruct/ap/internal/util/Strings.java#L217-L228

No other `similarWords` can happen if `sourceType.getPropertyReadAccessors().keySet()` is empty. If that is empty then there are no read properties. Perhaps we need to add that?

Reply:
Ah.. I understand you now.. but check my comment above..